CMakeLists.txt

```diff
@@ -8,8 +8,87 @@ cmake_policy(SET CMP0072 OLD) #not yet, would need to get pointers to all OpenGL
 set(OpenGL_GL_PREFERENCE LEGACY)
 #cmake_policy(SET CMP )
 
-project(Nabla LANGUAGES CXX C)
+option(NBL_BUILD_ANDROID "Build for Android?" OFF)
+
+if(NBL_BUILD_ANDROID)
+	# assert that cross-compile host OS is always Linux
+	if (NOT "${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Linux")
+		message(FATAL_ERROR "Cross-compile for Android is available on Linux only as a host OS!")
+	endif()
+
+    if(NOT DEFINED ENV{JAVA_HOME})
+        message(FATAL_ERROR "JAVA_HOME environment variable must be defined for Android build")
+    endif()
+    message(STATUS "Using JAVA_HOME = $ENV{JAVA_HOME}")
+
+    execute_process(COMMAND $ENV{JAVA_HOME}/bin/java -version
+        RESULT_VARIABLE _result
+        OUTPUT_VARIABLE _output
+        ERROR_VARIABLE _output
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_STRIP_TRAILING_WHITESPACE)
+    
+    if(NOT _result AND _output MATCHES "version \"([0-9]+).([0-9]+)")
+        message(STATUS "Java in JAVA_HOME is ${CMAKE_MATCH_1}.${CMAKE_MATCH_2}")
+    else()
+        message(STATUS "Java in JAVA_HOME is unknown version ${_output} ${_result}")
+    endif()
+
+    if(DEFINED ENV{ANDROID_HOME} AND EXISTS "$ENV{ANDROID_HOME}/build-tools")
+        set(ANDROID_SDK_ROOT_PATH "$ENV{ANDROID_HOME}")
+    elseif(DEFINED ENV{ANDROID_SDK_ROOT} AND EXISTS "$ENV{ANDROID_SDK_ROOT}/build-tools")
+        set(ANDROID_SDK_ROOT_PATH "$ENV{ANDROID_SDK_ROOT}")
+    elseif(DEFINED ENV{ANDROID_SDK} AND EXISTS "$ENV{ANDROID_SDK}/build-tools")
+        set(ANDROID_SDK_ROOT_PATH "$ENV{ANDROID_SDK}")
+    else()
+        message(FATAL_ERROR "Can't locate Android SDK, set ANDROID_HOME, ANDROID_SDK_ROOT or ANDROID_SDK")
+    endif()
+
+    message(STATUS "Using Android SDK found in ${ANDROID_SDK_ROOT_PATH}")
+
+    if(DEFINED ENV{ANDROID_NDK_HOME} AND EXISTS "$ENV{ANDROID_NDK_HOME}/build/cmake/android.toolchain.cmake")
+        set(ANDROID_NDK_ROOT_PATH "$ENV{ANDROID_NDK_HOME}")
+    elseif(DEFINED ENV{ANDROID_NDK_ROOT} AND EXISTS "$ENV{ANDROID_NDK_ROOT}/build/cmake/android.toolchain.cmake")
+        set(ANDROID_NDK_ROOT_PATH "$ENV{ANDROID_NDK_ROOT}")
+    elseif(DEFINED ENV{NDK_HOME} AND EXISTS "$ENV{NDK_HOME}/build/cmake/android.toolchain.cmake")
+        set(ANDROID_NDK_ROOT_PATH "$ENV{NDK_HOME}")
+    elseif(DEFINED ENV{ANDROID_NDK} AND EXISTS "$ENV{ANDROID_NDK}/build/cmake/android.toolchain.cmake")
+        set(ANDROID_NDK_ROOT_PATH "$ENV{ANDROID_NDK}")
+    else()
+        message(FATAL_ERROR "Can't locate Android NDK, set ANDROID_NDK_HOME, ANDROID_NDK_ROOT, NDK_HOME or ANDROID_NDK")
+    endif()
+
+    message(STATUS "Using Android NDK found in ${ANDROID_NDK_ROOT_PATH}")
+
+    set(CMAKE_TOOLCHAIN_FILE
+        "${ANDROID_NDK_ROOT_PATH}/build/cmake/android.toolchain.cmake"
+        CACHE STRING
+        "The Android toolchain file")
+
+    #option(STRIP_ANDROID_LIBRARY "Strip the resulting android library" OFF)
+
+    if(NOT ANDROID_PLATFORM)
+        set(ANDROID_PLATFORM "android-25")
+    endif()
```

Comment:
@Crisspl how far can you bump this and still have it run on ChromeOS

AFAIK all the chromebooks can upgrade to ver 91 which brigs to them Android 11 so API level 30

Reply:
TODO

Reply:
probably best to delegate to @AnastaZIuk